minichef/contracts/Strategy.sol

```diff
@@ -0,0 +1,458 @@
+                            _debtPayment
+                        );
+                    }
+                }
+            }
+        } else {
+            //serious loss should never happen but if it does lets record it accurately
+            _loss = debt - assets;
+        }
+
+        // we're done harvesting, so reset our trigger if we used it
+        forceHarvestTriggerOnce = false;
+    }
+
+
+
+    function prepareMigration(address _newStrategy) internal override {
+        uint256 stakedBalance = balanceOfStaked();
+        if (stakedBalance > 0) {
+            chef.withdraw(pid, stakedBalance, address(this));
+        }
+
+        // send our claimed emissionToken to the new strategy
+        emissionToken.safeTransfer(
+            _newStrategy,
+            emissionToken.balanceOf(address(this))
+        );
+    }
+
+
+    function adjustPosition(uint256 _debtOutstanding) internal override {
+        if (emergencyExit) {
+            return;
+        }
+        // send all of our want tokens to be deposited
+        uint256 toInvest = balanceOfWant();
+        // stake only if we have something to stake
+        if (toInvest > 0) {
+            chef.deposit(pid, toInvest, address(this));
+        }
+    }
+
+
+    // sell rewards token to targetTokens
+    function rewardToOptimal() internal {
+        uint256 rewardBalance = emissionToken.balanceOf(address(this));
+        if (rewardBalance > 0) {
+            // swap reward for optimal
+            _sell(
+                address(emissionToken), 
+                targetToken, 
+                rewardBalance
+            );
+        }
+    }
+
+
+    // swap rewards to target using assigned router
+    function _sell(
+        address _tokenFrom,
+        address _tokenTo,
+        uint256 _amount
+    ) internal {
+        address[] memory path;
+
+        if (_tokenFrom == address(wavax)) {
+            path = new address[](2);
+            path[0] = _tokenFrom;
+            path[1] = _tokenTo;
+        } else if (_tokenTo == address(wavax)) {
+            path = new address[](2);
+            path[0] = _tokenFrom;
+            path[1] = _tokenTo;
+        } else {
+            path = new address[](3);
+            path[0] = _tokenFrom;
+            path[1] = address(wavax);
+            path[2] = _tokenTo;
+        }
+
+        router.swapExactTokensForTokens(_amount, 0, path, address(this), now);
+    }
+
+
+    /* ========== SETTERS ========== */
+
+    ///@notice This allows us to manually harvest with our keeper as needed
+    function setForceHarvestTriggerOnce(bool _forceHarvestTriggerOnce)
+        external
+        onlyAuthorized
+    {
+        forceHarvestTriggerOnce = _forceHarvestTriggerOnce;
+    }
+
+    ///@notice When our strategy has this much credit, harvestTrigger will be true.
+    function setMinHarvestCredit(uint256 _minHarvestCredit)
+        external
+        onlyAuthorized
+    {
+        minHarvestCredit = _minHarvestCredit;
+    }
+
+}
```

Comment:
Is this strategy based on an existing strategy from yearn? Can we somehow review a diff of this strategy against the original one?

Reply:
Here is the strategy we used as example , [](https://github.com/dudesahn/Strategy0xDAOStaker/blob/main/contracts/Strategy0xDAOStaker.sol)

Reply:
@joshualyguessennd It's based off of this [one](https://github.com/dudesahn/Strategy0xDAOStaker/blob/main/contracts/Strategy0xDAOStaker.sol), right?

Can you make the diff and post it here?
We should also probably include this as part of the PR template if strategy is based off of a existing one